src/open_samus_returns_rando/files/levels/s070_area7.lua

```diff
@@ -889,6 +893,9 @@ function s070_area7.OnStartManicMinerBotDeathCutscene()
   Game.AddEntityToUpdateInCutscene("LE_ManicMinerWall")
   Game.AddEntityToUpdateInCutscene("LE_PowerUp_Powerbomb")
   Scenario.WriteToBlackboard("manicMinerDead", "b", true)
+  local PostFightPBMax = Blackboard.GetProp("PLAYER_INVENTORY", "ITEM_WEAPON_POWER_BOMB_MAX")
+  Blackboard.SetProp("PLAYER_INVENTORY", "ITEM_WEAPON_POWER_BOMB", "f", PBLauncher)
+  Blackboard.SetProp("PLAYER_INVENTORY", "ITEM_WEAPON_POWER_BOMB_MAX", "f", PostFightPBMax + PreFightPBMax)
```

Comment:
The launcher does not exist in the vanilla game like that. As soon as we have `ITEM_WEAPON_POWER_BOMB_MAX > 0` you can use PBs.

What item we grant is modified depending if `ITEM_WEAPON_POWER_BOMB` is in the inventory or not.
The item sent or picked up is always `ITEM_POWER_BOMB_TANKS`. (no idea why the plural made it into the code).
So, it is kept as `ITEM_POWER_BOMB_TANKS` if `ITEM_WEAPON_POWER_BOMB = 0` else it is changed to `ITEM_WEAPON_POWER_BOMB_MAX `.
See:
https://github.com/randovania/open-samus-returns-rando/blob/195969116cce52f8ac7de594d8832997ee7d9716/src/open_samus_returns_rando/files/templates/randomizer_powerup.lua#L195-L200

That makes the whole logic cursed especially if you take into account you could receive the main item during the fight (or a second one).

AFAIK you need to get `ITEM_WEAPON_POWER_BOMB` after the fight and sum it up with `PBLauncher`. If that is ` >=1` we either had the launcher before or received it during the fight. 
`PostFightPBMax` needs to be summed up with with `ITEM_POWER_BOMB_TANKS` (a.k.a. "locked" PB tanks) and line 898 needs to be conditional: 
1. Use `ITEM_WEAPON_POWER_BOMB_MAX` if `"sum of launchers" >= 1` and reset `ITEM_POWER_BOMB_TANKS` to 0
2. else keep them locked and use  `ITEM_POWER_BOMB_TANKS` .

Reply:
Wouldn't getting a Launcher mid

Reply:
Yeah, it would break in the meaning of that the PBs gets activated.
But it would also break much more: The `ITEM_WEAPON_POWER_BOMB` would be set to pre-fight value, which was 0. Due to the linked logic from above, you won't get any new tanks because if you pick them up, they will be locked as you don't have the main item.

Example:
```
---
Pre-fight
ITEM_WEAPON_POWER_BOMB = 0,
ITEM_WEAPON_POWER_BOMB_MAX = 0,
ITEM_POWER_BOMB_TANKS = 5,
PBLauncher = 0
PreFightPBMax = 0
---

---
During fight (receive main item => unlocks the tanks):
ITEM_WEAPON_POWER_BOMB = 1,
ITEM_WEAPON_POWER_BOMB_MAX = 5,
ITEM_POWER_BOMB_TANKS = 0,
PBLauncher = 0
PreFightPBMax = 0
---

---
After fight:
ITEM_WEAPON_POWER_BOMB = 0, (was reset to PBLauncher value)
ITEM_WEAPON_POWER_BOMB_MAX = 5,
ITEM_POWER_BOMB_TANKS = 0,
PBLauncher = 0
PreFightPBMax = 0

=> 5 PBs are usable, no more are added because main item was reset to 0
---
```

Reply:
Oh I see the issue. So I should change both or just the launcher then?

Reply:
Don't know what you mean by "both" here. But basically it should be this(?):

> AFAIK you need to get `ITEM_WEAPON_POWER_BOMB` after the fight and sum it up with `PBLauncher`. If that is ` >=1` we either had the launcher before or received it during the fight. `PostFightPBMax` needs to be summed up with with `ITEM_POWER_BOMB_TANKS` (a.k.a. "locked" PB tanks) and line 898 needs to be conditional:
> 
> 1. Use `ITEM_WEAPON_POWER_BOMB_MAX` if `"sum of launchers" >= 1` and reset `ITEM_POWER_BOMB_TANKS` to 0
> 2. else keep them locked and use  `ITEM_POWER_BOMB_TANKS` .

Reply:
I hope this is what you meant